_clients/java-rest-high-level.md

```diff
@@ -25,7 +22,33 @@ To start using the OpenSearch Java high-level REST client, ensure that you have
 
 You can now start your OpenSearch cluster. The OpenSearch 1.x high-level REST client works with the 1.x versions of OpenSearch.
 
-The following example uses credentials that come with the default OpenSearch configuration. If you’re using the high-level REST client with your own OpenSearch cluster, be sure to change the code to use your own credentials.
+## Security
+
+This code example uses basic credentials that come with the default OpenSearch configuration. If you’re using the OpenSearch Java high-level REST client with your own OpenSearch cluster, be sure to change the code to use your own credentials.
+{: .note}
+
+Before you can securely connect to an OpenSearch cluster, you must first add your root certificates to a truststore. If you don't already have a custom truststore, the following command creates a truststore and adds in a certificate.
+
+```bash
+keytool -import <path-to-cert> -alias <alias-to-call-cert> -keystore <truststore-name>
+```
+
+You can now point your Java client to the truststore and set basic authentication credentials that can access a secure cluster.
+
+```java
+//Point to keystore with appropriate certificates for security.
+System.setProperty("javax.net.ssl.trustStore", "/full/path/to/keystore");
+System.setProperty("javax.net.ssl.trustStorePassword", "password-to-keystore");
+
+//Establish credentials to use basic authentication.
+//Only for demo purposes. Don't specify your credentials in code.
+final CredentialsProvider credentialsProvider = new BasicCredentialsProvider();
+credentialsProvider.setCredentials(AuthScope.ANY,
+  new UsernamePasswordCredentials("admin", "admin"));
+```
```

Comment:
Can avoid duplicating the code here since it's already exists in the sample code.

Reply:
I can add in a `(refer to sample code below on how to do so)`.